cohort-cli/src/main/java/com/ibm/cohort/cli/MeasureCLI.java

```diff
@@ -79,6 +90,18 @@
 				"--define-return-option" }, description = "Specify define return option for evaluated define statements on measure report. Defaults to NONE.  Valid options are (ALL|BOOLEAN|NONE).")
 		private DefineReturnOptions defineReturnOption = DefineReturnOptions.NONE;
 
+		@Parameter(names = { "--enable-retrieve-cache" }, description = "Enable the use of the retrieve cache.")
+		private boolean enableRetrieveCache = false;
+
+		@Parameter(names = { "--max-retrieve-cache-size" }, description = "The maximum size the retrieve cache can grow before evictions begin.")
+		private int maxRetrieveCacheSize = 1_000;
+
+		@Parameter(names = { "--retrieve-cache-expire-on-write" }, description = "The amount of time after last write (in seconds) before a retrieve cache entry is evicted.")
+		private int retrieveCacheExpireOnWrite = 300;
+
+		@Parameter(names = { "--enable-retrieve-cache-statistics" }, description = "Enable retrieve cache statistic recording via JMX.")
+		private boolean enableRetrieveCacheStatistics = false;
```

Comment:
Rather than have options to our program for all the various possible configuration settings of the cache, did you consider making these configurable via a configuration file? For example, something like [this reference conf file](https://github.com/ben-manes/caffeine/blob/master/jcache/src/main/resources/reference.conf.)

Reply:
That configuration file is specifically for Caffeine.
It's meant to be put on the classpath and automatically read in by Typesafe's config library instead of passed in as a parameter.
That makes it hard for users to use their own config file with our CLI program (not sure if they should regardless).
They should be fine to use their own configs within their own distributions though.

- Are you looking for me to leverage the actual configfile implementation for Caffeine, or just create my own json format for configuring Caffeine myself?
- Do you want the configuration file to be implementation agnostic?  Identifying what properties make sense in an agnostic configuration file (and implementing the logic around it) is big enough to be another ticket.

I initially hit concurrency issues when I was using the typesafe config file to create new caches, but didn't think to use it to set the default values.  I can try that if desired.

Reply:
Actually, it looks like users can override the config file with something like `-Dconfig.file=path/to/config-file` based on the typesafe docs.
I could try to provide a config file with some altered default settings.
I'm guessing we'll still want `--enable-cache` to remain though correct?

Reply:
My general thought here is that cache configuration is more complex than these three properties. These might be enough for a basic setup, but I was imagining a world where we would want to enable more complex cache configurations without having to add flags to our CLI for each specific setting. I know that most cache providers provide a configuration file format that can be used to externalize configuration. The above referenced file is the defaults file for caffeine and they seem to expect that an application would have an application.conf file. That is the one I would expect us to point to in some way.

Reply:
I think I was able to (somewhat) cleanly utilize the Typesafe/Lightbend config library after [this commit](https://github.com/Alvearie/quality-measure-and-cohort-service/pull/83/commits/933c307e989a51f39d2b1cd0a378e3d348f65568).
I had to do some work in the uberjar configuration to ensure all values in all `reference.conf` files remain intact, but everything else seemed obvious.
Let me know if this addresses your concerns.

Reply:
Looks ok to me.